docs/NNNN-hashed-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,242 @@
+# OCFL Community Extension NNNN: Hashed Truncated N-tuple Storage Layout
```

Comment:
I wonder whether "truncated" is helpful in the name. Nothing is really truncated, there may just be a remainder that isn't split into tuples